src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -528,6 +530,80 @@ public ImmutableList<Folder> listFolders(UUID workspaceId) {
     return ImmutableList.copyOf(allFolders);
   }
 
+  public Folder createFolder(
+      UUID workspaceId,
+      String displayName,
+      String description,
+      UUID parentId,
+      Map<String, String> properties) {
+    return callWithRetries(
+        () ->
+            new FolderApi(apiClient)
+                .createFolder(
+                    new CreateFolderRequestBody()
+                        .displayName(displayName)
+                        .description(description)
+                        .parentFolderId(parentId)
+                        .properties(PropertiesUtils.stringMapToProperties(properties)),
+                    workspaceId),
+        "Error creating folder");
+  }
+
+  /**
+   * Delete folder and all the sub-folders and resources within it.
+   *
+   * <p>Delete folder is an async endpoint. Resource deletion may take a long time.
+   *
+   * @throws InterruptedException
+   */
+  public void deleteFolder(UUID workspaceId, UUID folderId) throws InterruptedException {
+
+    var folderApi = new FolderApi(apiClient);
+    var jobResult =
+        callWithRetries(
+            () -> folderApi.deleteFolderAsync(workspaceId, folderId), "Error deleting folder");
+
+    var jobId = jobResult.getJobReport().getId();
+    // poll the result endpoint until the job is no longer RUNNING
+    jobResult =
+        HttpUtils.pollWithRetries(
```

Comment:
Why not return immediately with a message that it being deleted? 

this is the same issue as in sagemaker notebooks.

Reply:
i copied the workspace deletion pattern